src/utils/format_utils.hxx

```diff
@@ -71,4 +71,31 @@ char* unsigned_integer_to_buffer_hex(int value, char* buffer);
 ///
 string mac_to_string(uint8_t mac[6], bool colons = true);
 
+/// Formats an IPv4 address to string.
+///
+/// @param ip a 4-byte array storing the IPv4 address. ip[3] will be printed
+/// at the beginning.
+/// @param dots true to print dots, else false to exclude the dot
+/// seperators
+///
+/// @return a string containing a dot-separated hexadecimal printout of the
```

Comment:
it seems this is not hexadecimal but decimal.

for this reason i'm not sure what is the use of dot == false. When do you call it with !dot?